File: samples/scala/deadbolt-usage-scala/app/controllers/PatternController.scala
package controllers

import security.MyDeadboltHandler
import play.api.mvc.{Action, Controller}
import views.html.accessOk
import be.objectify.deadbolt.scala.DeadboltActions
import be.objectify.deadbolt.core.PatternType

/**
 *
 * @author Steve Chaloner (steve@objectify.be)
 */
object PatternController extends Controller with DeadboltActions
{
  def printersEdit = Pattern("printers.edit", PatternType.EQUALITY, new MyDeadboltHandler) {
                   Action {
                     Ok(accessOk())
                   }
                 }

  def printersFoo = Pattern("printers.foo", PatternType.EQUALITY, new MyDeadboltHandler) {
                   Action {
                     Ok(accessOk())
                   }
                 }

  def printersRegex = Pattern("(.)*\\.edit", PatternType.REGEX, new MyDeadboltHandler) {
                   Action {
                     Ok(accessOk())
                   }
                 }
}

File: samples/java/deadbolt-usage/app/controllers/PatternController.java
/*
 * Copyright 2012 Steve Chaloner
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package controllers;

import be.objectify.deadbolt.core.PatternType;
import be.objectify.deadbolt.java.actions.Pattern;
import play.mvc.Controller;
import play.mvc.Result;
import views.html.accessOk;

/**
 * @author Steve Chaloner (steve@objectify.be)
 */
public class PatternController extends Controller
{
    @Pattern("printers.edit")
    public static Result editPrinter()
    {
        return ok(accessOk.render());
    }

    @Pattern("printers.detonate")
    public static Result detonatePrinter()
    {
        return ok(accessOk.render());
    }

    @Pattern(value = "(.)*\\.edit", patternType = PatternType.REGEX)
    public static Result editPrinterRegex()
    {
        return ok(accessOk.render());
    }
}
